### ISSUE TITLE
Embed API 以外の CSS が邪魔されないように仕組みを考える

### ISSUE THREAD

**Issue by keichan34** [geolonia/embed#226]
埋め込み先サイトのCSSにより表示が崩れるケースが有る #224 

* web components で対応できる？ #216 ?
* shadow DOM で対応できる？mapbox-gl-js はデフォルトで[対応していない](https://github.com/mapbox/mapbox-gl-js/issues/7814)だが、CSSがうまく読み込まれていないだけみたいなので、微調整でshadow DOMで実装できるかな？

**Comment by keichan34** [comment]
https://codepen.io/keichan34/pen/ZEKqeEj?editors=1111

stylesheetをshadow DOM内に読み込めば行けそう

**Comment by kochizufan** [comment]
突然すみません、購読してるので流れてきましたので...

ご参考ですが、Maplatでは、使ってるcssを全部lessで読み込んで.maplatクラスの下に編成して、
Maplatを構成するdivにmaplatクラスを加えることで、外部から内部にも、内部から外部にもcssの影響を与えないようにしています。

https://github.com/code4history/Maplat/blob/master/less/ui.less

**Comment by keichan34** [comment]
@kochizufan コメントありがとうございました！

実は現在クラスは全部一応プレフィックスされているが、エンベッド先のサイトに `div` や `div > :first-child` などのクラスを指定しないCSSが読み込まれているの場合、レイアウトが崩れてしまいます。。

**Comment by miya0001** [comment]
これもしShadow DOMで対処するならたとえばWordPressのプラグイン内でやるとかにして、Embed API内での対処は避けたいです。
危惧しているのは万が一MapLibre側でこれに対する対処が入るなどのケースで、互換性とかの心配事が増えすぎるなと。

見ればすぐに分かることなので、とりあえず今はサポートとかドキュメンテーションとかで対処しましょう。
もしどうしても対処するならMapLibreにプルリクを送りましょう。

**Comment by keichan34** [comment]
そうですね。embed API 内で shadow DOM で対応するにしても最初からデフォルトにする（わざわざCSSで内部までスタイリングしているケースがあると思いまして）想定でなく、少なくとも `data-shadow-dom="on"` 的なスイッチの裏で実装しようと思っていました。

WordPressのプラグインに入れるのはいいかもね。
